packages/kbn-check-mappings-update-cli/src/mappings_additions/extract_field_lists_from_plugins.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import ChildProcess from 'child_process';
+import { Readable } from 'stream';
+import * as Rx from 'rxjs';
+import { REPO_ROOT } from '@kbn/repo-info';
+import { SomeDevLog } from '@kbn/some-dev-log';
+import { observeLines } from '@kbn/stdio-dev-helpers';
+import type { Result } from './extract_field_lists_from_plugins_worker';
+
+function routeToLog(readable: Readable, log: SomeDevLog, level: 'debug' | 'error') {
+  return observeLines(readable).pipe(
+    Rx.tap((line) => {
+      log[level](line);
+    }),
+    Rx.ignoreElements()
+  );
+}
+
+/**
+ * Run a worker process that starts the core with all plugins enabled and sends back the
+ * saved object mappings for all plugins. We run this in a child process so that we can
+ * harvest logs and feed them into the logger when debugging.
+ */
+export async function extractFieldListsFromPlugins(log: SomeDevLog): Promise<Result> {
+  log.info('Loading core with all plugins enabled so that we can get all savedObject mappings...');
+
+  const fork = ChildProcess.fork(require.resolve('./extract_field_lists_from_plugins_worker.ts'), {
+    execArgv: ['--require=@kbn/babel-register/install'],
+    cwd: REPO_ROOT,
+    stdio: ['ignore', 'pipe', 'pipe', 'ipc'],
+  });
```

Comment:
I was wondering why the other check was using a worker, and decided to go without it, and then I understood: starting the server in this awkward state keeps some open handles, forcing the process to never terminate, so we're effectively forced to fork.

I didn't bother factorizing the fork creating code yet. I will probably if/when we need to add a third check.

Reply:
> I didn't bother factorizing the fork creating code yet. I will probably if/when we need to add a third check.

Good choice, 2 instances of the same/ similar code is fine, a third is a pattern.

Reply:
> starting the server in this awkward state keeps some open handles, forcing the process to never terminate, so we're effectively forced to fork.

We need to add this info somewhere in the package to prevent others from falling into the same trap.

Reply:
Yeah makes sense, I will add a comment around the place we're invoking the workers